### PR TITLE
google maps api integration

### PR DESCRIPTION
**Pull request by cathli66**
# Summary
Integrating Google Maps API into CornellGo's app for our gameplay.
Does not include the Google Maps API key

- [x] Updated Android and iOS build files to include Google Maps API class
- [x] Changed gameplay_map.dart from Flutter map (old) to Google Maps (new)
- [x] Tested Android Map SDK integration
- [x] Display and zoom to user's current location
- [x] Draw some random circle around a challenge for user testing
- [x] Move maps into gameplay page
- [x] Test Map SDK integration on iPhone

## Remaining TODOs:
- [ ] Connect with backend

## Test Plan
Running on emulators or real devices. I need someone else to test iOS for me since I don't have XCode.

## Notes
Does not include API key!! Reach out to me (through Slack) to get the API key when testing this PR.

### CODE REVIEW

**Review by neketka — DISMISSED** [review]
Some comments about commented code + possible example code

File: game/lib/api/geopoint.dart [hunk 1]
```diff
@@ -59,4 +61,40 @@ class GeoPoint {
...
+      locationSettings = AndroidSettings(
+          accuracy: LocationAccuracy.high,
+          distanceFilter: 100,
+          forceLocationManager: true,
+          intervalDuration: const Duration(seconds: 10),
+          //(Optional) Set foreground notification config to keep the app alive
+          //when going to the background
+          foregroundNotificationConfig: const ForegroundNotificationConfig(
+            notificationText:
+                "Example app will continue to receive your location even when you aren't using it",
```
[the remark this answers is not in the document]
Comment (cathli66) [reply game/lib/api/geopoint.dart:78]: fixed - thanks for catching!

File: game/lib/gameplay/gameplay_map.dart [hunk 2]
```diff
@@ -12,20 +15,123 @@ class GameplayMap extends StatefulWidget {
...
 
 class _GameplayMapState extends State<GameplayMap> {
-  final mapController = MapController();
+  // AndroidMapRenderer mapRenderer = AndroidMapRenderer.platformDefault;
+
+  // final mapController = MapController();
+  late Completer<GoogleMapController> mapCompleter = Completer();
+
+  final LatLng _center = const LatLng(-33.86, 151.20);
+  // final LatLng _center = const LatLng(40.00, -70.00);
```
[the remark this answers is not in the document]
Comment (cathli66) [reply game/lib/gameplay/gameplay_map.dart:24]: Either way, it jumps almost immediately to the user's current location so I feel like it shouldn't matter too much, as long as the default center is somewhere in Cornell.

File: game/lib/gameplay/gameplay_map.dart [hunk 3]
```diff
@@ -12,20 +15,123 @@ class GameplayMap extends StatefulWidget {
 }
 
 class _GameplayMapState extends State<GameplayMap> {
-  final mapController = MapController();
+  // AndroidMapRenderer mapRenderer = AndroidMapRenderer.platformDefault;
```
[the remark this answers is not in the document]
Comment (cathli66) [reply game/lib/gameplay/gameplay_map.dart]: I still need to test on iOS, but yea having this code here could break it. I've removed it, shouldn't be there anyways

**Review by jasmineee-li — COMMENTED** [review]
Ran on my computer, looks good! I liked how you added comments and TODOs in the code. I can help test iOS with XCode on my end.

**Review by cgu2020 — COMMENTED** [review]
Can confirm that the maps do not work on iOS. Using GoogleMapsFlutterAndroid strictly in main.dart which might be a reason. Let me know when you fix this so I can test it on my simulator.

**Review by 3TTemi — APPROVED** [review]
Tested going into Google Maps from a challenge on IOS. Recentering worked so everything looks good!

**Pull request merged**